proto_text.c

```diff
@@ -105,6 +105,16 @@ static void _finalize_mset(conn *c, enum store_item_type ret) {
                 META_CHAR(p, 'c');
                 p = itoa_u64(c->cas, p);
                 break;
+            case 's':
+                // Get final item size, ie from append/prepend
+                META_CHAR(p, 's');
+                // If the size changed during append/prepend
+                if (nbytes != 0) {
+                    p = itoa_u32(nbytes-2, p);
+                } else {
+                    p = itoa_u32(it->nbytes-2, p);
+                }
```

Comment:
noob question: can we always just rely on this?

Reply:
at a high level it seems like we are passing a int pointer to `do_item_store` where we update the final length of the `it` based on `it->nbytes`, can we skip passing that pointer and just use the value?

Reply:
No, sorry that code is insane. `it` is being overwritten by `new_it` inside do_store_item, and then it's lost forever. I had to add the nbytes bit to get that data out of the function.

I added the conditional in weird places because I'm trying to avoid having the checks for "did nbytes change from do_store_item?" exist outside of append/prepend and `s` flag being used

Reply:
only other reasonable option is to make the passed in item ptr be **item and replace it if it's replaced, but that's going to be even less legible + even more fiddling around refcounts than it's already doing.

Reply:
I get the first part, but don't follow the later. 

>I added the conditional in weird places because I'm trying to avoid having the checks for "did nbytes change from do_store_item?" exist outside of append/prepend and s flag being used

wouldn't the `it` always contain the latest appended/prepended item after execution of `do_item_store`?

Reply:
it is passed into do_store_item, not out.

Reply:
in append/prepend do_store_item takes an input item, looks up the original item, then merges them together. the item available afterward was still the input item, so without this code it just returns the size of the append command itself.

Reply:
ack - that makes sense!